File: lib/src/domain/models/query_element.dart
import 'package:equatable/equatable.dart';

import 'package:query_wizard/domain.dart';

class QueryElement extends Equatable {
  const QueryElement({
    required this.name,
    this.alias,
    required this.type,
    this.parent,
    required this.elements,
  });

  final String name;
  final String? alias;
  final QueryElementType type;
  final QueryElement? parent;
  final List<QueryElement> elements;

  @override
  List<Object> get props => [
        name,
        alias ?? '',
        type,
        parent ?? '',
        elements,
      ];
}

File: server/src/QueryWizard.Domain/src/Entities/QueryElement.cs
using System.Collections.Generic;
using QueryWizard.Domain.Enums;

namespace QueryWizard.Domain.Entities
{
    public class QueryElement
    {
        public string Id { get; set; }

        public string Name { get; set; }

        public string Alias { get; set; }

        public QueryElementType Type { get; set; }

        public QueryElement Parent { get; set; }

        public List<QueryElement> Elements { get; set; }
    }
}
